boards/raspberry_pi_pico/src/main.rs

```diff
@@ -0,0 +1,431 @@
+    ReferenceAuxiliaryClockSource, ReferenceClockSource, RtcAuxiliaryClockSource,
+    SystemAuxiliaryClockSource, SystemClockSource, UsbAuxiliaryClockSource,
+};
+use rp2040::gpio::{GpioFunction, RPGpio, RPGpioPin};
+use rp2040::resets::Peripheral;
+use rp2040::timer::RPTimer;
+mod io;
+
+mod flash_bootloader;
+
+/// Allocate memory for the stack
+#[no_mangle]
+#[link_section = ".stack_buffer"]
+pub static mut STACK_MEMORY: [u8; 0x1000] = [0; 0x1000];
+
+// Manually setting the boot header section that contains the FCB header
+#[used]
+#[link_section = ".flash_bootloader"]
+static FLASH_BOOTLOADER: [u8; 256] = flash_bootloader::FLASH_BOOTLOADER;
+
+// State for loading and holding applications.
+// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: kernel::procs::PanicFaultPolicy = kernel::procs::PanicFaultPolicy {};
+
+// Number of concurrent processes this platform supports.
+const NUM_PROCS: usize = 4;
+
+static mut PROCESSES: [Option<&'static dyn kernel::procs::Process>; NUM_PROCS] = [None; NUM_PROCS];
+
+static mut CHIP: Option<&'static Rp2040<Rp2040DefaultPeripherals>> = None;
+
+/// Supported drivers by the platform
+pub struct RaspberryPiPico {
+    ipc: kernel::ipc::IPC<NUM_PROCS>,
+    console: &'static capsules::console::Console<'static>,
+    alarm: &'static capsules::alarm::AlarmDriver<
+        'static,
+        VirtualMuxAlarm<'static, rp2040::timer::RPTimer<'static>>,
+    >,
+    gpio: &'static capsules::gpio::GPIO<'static, RPGpioPin<'static>>,
+    led: &'static capsules::led::LedDriver<'static, LedHigh<'static, RPGpioPin<'static>>>,
+    adc: &'static capsules::adc::AdcVirtualized<'static>,
+    temperature: &'static capsules::temperature::TemperatureSensor<'static>,
+}
+
+impl Platform for RaspberryPiPico {
+    fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
+    where
+        F: FnOnce(Option<&dyn kernel::Driver>) -> R,
+    {
+        match driver_num {
+            capsules::console::DRIVER_NUM => f(Some(self.console)),
+            capsules::alarm::DRIVER_NUM => f(Some(self.alarm)),
+            capsules::gpio::DRIVER_NUM => f(Some(self.gpio)),
+            capsules::led::DRIVER_NUM => f(Some(self.led)),
+            kernel::ipc::DRIVER_NUM => f(Some(&self.ipc)),
+            capsules::adc::DRIVER_NUM => f(Some(self.adc)),
+            capsules::temperature::DRIVER_NUM => f(Some(self.temperature)),
+            _ => f(None),
+        }
+    }
+}
+
+/// Entry point used for debuger
+///
+/// When loaded using gdb, the Raspberry Pi Pico is not reset
+/// by default. Without this function, gdb sets the PC to the
+/// beginning of the flash. This is not correct, as the RP2040
+/// has a more complex boot process.
+///
+/// This function is set to be the entry point for gdb and is used
+/// to send the RP2040 back in the bootloader so that all the boot
+/// sqeuence is performed.
+#[no_mangle]
+#[naked]
+pub unsafe extern "C" fn reset() {
```

Comment:
```suggestion
pub unsafe extern "C" fn jump_to_bootloader() {
```
?
Or is the name `reset` required?

Reply:
done